pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -212,73 +211,17 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 
 	proxyInstances := node.ServiceInstances
 	noneMode := node.GetInterceptionMode() == model.InterceptionNone
-	listeners := make([]*xdsapi.Listener, 0)
-
-	actualWildcard, actualLocalHostAddress := getActualWildcardAndLocalHost(node)
+	_, actualLocalHostAddress := getActualWildcardAndLocalHost(node)
 
 	builder := NewListenerBuilder(node)
 	if mesh.ProxyListenPort > 0 {
-		// Notes that the the else branch works for both split and non-split cases.
-		// TODO(silentdai): remove `if` branch once split behavior is well verified
-		if !node.IsInboundCaptureAllPorts() {
-			inbound := configgen.buildSidecarInboundListeners(env, node, push, proxyInstances)
-			outbound := configgen.buildSidecarOutboundListeners(env, node, push, proxyInstances)
-
-			builder.inboundListeners = inbound
-			builder.outboundListeners = outbound
-			builder.managementListeners = configgen.generateManagementListeners(node, noneMode, env, listeners)
-
-			tcpProxy := &tcp_proxy.TcpProxy{
-				StatPrefix:       util.BlackHoleCluster,
-				ClusterSpecifier: &tcp_proxy.TcpProxy_Cluster{Cluster: util.BlackHoleCluster},
-			}
-
-			if isAllowAny(node) {
-				// We need a passthrough filter to fill in the filter stack for orig_dst listener
-				tcpProxy = &tcp_proxy.TcpProxy{
-					StatPrefix:       util.PassthroughCluster,
-					ClusterSpecifier: &tcp_proxy.TcpProxy_Cluster{Cluster: util.PassthroughCluster},
-				}
-			}
-
-			setAccessLog(env, node, tcpProxy)
-
-			var transparent *google_protobuf.BoolValue
-			if node.GetInterceptionMode() == model.InterceptionTproxy {
-				transparent = proto.BoolTrue
-			}
-
-			filter := listener.Filter{
-				Name: xdsutil.TCPProxy,
-			}
-
-			if util.IsXDSMarshalingToAnyEnabled(node) {
-				filter.ConfigType = &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(tcpProxy)}
-			} else {
-				filter.ConfigType = &listener.Filter_Config{Config: util.MessageToStruct(tcpProxy)}
-			}
-
-			// add an extra listener that binds to the port that is the recipient of the iptables redirect
-			builder.virtualListener = &xdsapi.Listener{
-				Name:           VirtualListenerName,
-				Address:        util.BuildAddress(actualWildcard, uint32(mesh.ProxyListenPort)),
-				Transparent:    transparent,
-				UseOriginalDst: proto.BoolTrue,
-				FilterChains: []listener.FilterChain{
-					{
-						Filters: []listener.Filter{filter},
-					},
-				},
-			}
-		} else {
-			// Any build order change need a careful code review
-			builder = NewListenerBuilder(node).
-				buildSidecarInboundListeners(configgen, env, node, push, proxyInstances).
-				buildSidecarOutboundListeners(configgen, env, node, push, proxyInstances).
-				buildManagementListeners(configgen, env, node, push, proxyInstances).
-				buildVirtualListener(env, node).
-				buildVirtualInboundListener(env, node)
-		}
+		// Any build order change need a careful code review
```

Comment:
I assume you're removing the if block because the split behavior has been well verified (as the comment indicates)?

Reply:
Yep.. even if not, we would still be shipping an additional no-op virtual listener

Reply:
I am seeing 5~ish changes since the builder. Could you confirm if the builder is updated?

Reply:
There is no change in the build order. Also, the build order wont affect backward compat for that matter.